test/assets/unrunnableTestCases.csv

```diff
@@ -2656,77 +2636,64 @@ K2-DirectConElemWhitespace-5,Error: Not implemented: only module imports, namesp
+ForExprType071,AssertionError: expected [Function] to throw an error
 ForExprType072,AssertionError: expected [Function] to throw an error
-ForExprType073,AssertionError: expected [Function] to throw error matching /XPTY0004/ but got 'XPST0017: Function data with arity of 1 not registered. Did you mean "Q{http://www.w3.org/2001/XMLSchema}date (xs:anyAtomicType?)"?'
+ForExprType073,AssertionError: expected [Function] to throw an error
 K2-ForExprWith-1,AssertionError: expected [Function] to throw an error
 FunctionCall-009,Error: XPST0003: Unable to parse: "(         declare function local:f($in as xs:decimal*) as xs:decimal {sum($in, 0.0)};         local:f(xs:NMTOKENS('1 1.2 1.3 1.4')!xs:untypedAtomic(.))       ) instance of xs:decimal". Expected " ", "!", "!=", "(", "(:", ")", "*", "+", ",", "-", "/", "//", "<", "<<", "<=", "=", "=>", ">", ">=", ">>", "[", "\n", "\r", "\t", "and", "cast", "castable", "div", "eq", "except", "ge", "gt", "idiv", "instance", "intersect", "is", "le", "lt", "mod", "ne", "or", "to", "treat", "union", "|", or "||" but "f" found. (         declare [Error is around here]function local:f($in as xs:decimal*) as xs:decimal {sum($in, 0.0)};         local:f(xs:NMTOKENS('1 1.2 1.3 1.4')!xs:untypedAtomic(.))       ) instance of xs:decimal
 FunctionCall-010,Error: XPST0081: Can not cast: type xs:NMTOKENS is unknown.
 FunctionCall-011,Error: XPST0081: Can not cast: type xs:NMTOKENS is unknown.
 FunctionCall-012,AssertionError: expected [Function] to throw error matching /XPTY0004/ but got 'XPST0081: Can not cast: type xs:NMTOKENS is unknown.'
 FunctionCall-013,Error: XPST0003: Unable to parse: "(         declare function local:f($a as xs:integer, $b as xs:integer) as xs:integer {           data(<a>{$a}{$b}</a>)         };         local:f(12, 34)       ) instance of xs:integer". Expected " ", "!", "!=", "(", "(:", ")", "*", "+", ",", "-", "/", "//", "<", "<<", "<=", "=", "=>", ">", ">=", ">>", "[", "\n", "\r", "\t", "and", "cast", "castable", "div", "eq", "except", "ge", "gt", "idiv", "instance", "intersect", "is", "le", "lt", "mod", "ne", "or", "to", "treat", "union", "|", or "||" but "f" found. (         declare [Error is around here]function local:f($a as xs:integer, $b as xs:integer) as xs:integer {           data(<a>{$a}{$b}</a>)         };         local:f(12, 34)       ) instance of xs:integer
-FunctionCall-014,Error: XPST0017: Function data with arity of 1 not registered. Did you mean "Q{http://www.w3.org/2001/XMLSchema}date (xs:anyAtomicType?)"?
+FunctionCall-014,AssertionError: Expected XPath          let $f := function($a as xs:integer, $b as xs:integer) as xs:integer {           data(<a>{$a}{$b}</a>)         }         return $f(12, 34)        to resolve to something of type xs:integer: expected false to be true
```

Comment:
I think the data function should cast the return type or the like? @DrRataplan could you clarify the approach for this, this seems to be the same issue you solved some time back (return types for functions).

Reply:
yeah it should... I don't think this is related though